src/plugins/intel_cpu/src/cpu_tensor.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+#include <map>
+
+#include "cpu_tensor.h"
+#include "ie_blob.h"
+#include "ie_ngraph_utils.hpp"
+#include "openvino/runtime/iremote_tensor.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+//For the beginning probably one contructor should be enough
+Tensor::Tensor(MemoryPtr memptr) : m_memptr{memptr} {
+    OPENVINO_ASSERT(m_memptr != nullptr);
+
+    // only support plain data format ncsp.
+    auto memdesc = m_memptr->getDescPtr();
+    OPENVINO_ASSERT(memdesc->hasLayoutType(LayoutType::ncsp), "intel_cpu::Tensor only supports memory with ncsp layout.");
+    OPENVINO_ASSERT(m_memptr->getDescPtr()->isDefined(), "intel_cpu::Tensor requires memory with defined descriptor.");
+
+    m_element_type = InferenceEngine::details::convertPrecision(memdesc->getPrecision());
+}
+
+void Tensor::set_shape(ov::Shape new_shape) { // this is not as simple as it looks
+// if the ov::Shape is defined, then call cloneWithNewDims to the Memory object memory desc
+// if ov::Shape is undefined (dynamic), we probably have to reintialize the internal Memory object
+// using plain memory format
+    auto desc = m_memptr->getDescPtr();
+    const auto newdesc = desc->cloneWithNewDims(new_shape, true);
+    m_memptr->redefineDesc(newdesc);
+}
+
+const ov::element::Type& Tensor::get_element_type() const { // trivial -> simply convert the precision type
+    return m_element_type;
+}
+
+const ov::Shape& Tensor::get_shape() const { //trivial -> convert cpu Shape to ov::Shape
+    auto& shape = m_memptr->getDescPtr()->getShape();
+    OPENVINO_ASSERT(shape.isStatic(), "intel_cpu::Tensor requires memory initialized with static shape.");
+
+    std::lock_guard<std::mutex> guard(m_lock);
+    m_shape = shape.toPartialShape().to_shape();
+    return m_shape;
```

Comment:
What is the sense to save the shape inside the object, `m_shape` is not used anyhow? Why cant we simply `return shape.toPartialShape().to_shape();`? Also it allow to avoid using the lock. Also probably we can skip the toPartialShape() stage, and initialize the ov::Shape from static shape, something like that `return {shape.getStaticDims()}`

Reply:
done

Reply:
Yes, does make sense, didn't notice that the return type is a reference. However using `toPartialShape` is still an unnecessary step. It is better to use `m_shape = ov::Shape{shape.getStaticDims()}` as I've mentioned above.

Reply:
done